modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISortRelevantObjectEntryModelListener.java

```diff
@@ -51,21 +58,55 @@ private void _validate(ObjectEntry objectEntry) {
 		try {
 			Map<String, Serializable> values = objectEntry.getValues();
 
+			long apiEndpointId = (long)values.get(
+				"r_apiEndpointToAPISorts_c_apiEndpointId");
+
 			if (!_objectEntryHelper.isValidObjectEntry(
-					(long)values.get("r_apiEndpointToAPISorts_c_apiEndpointId"),
-					"L_API_ENDPOINT")) {
+					apiEndpointId, "L_API_ENDPOINT")) {
 
 				throw new ObjectEntryValuesException.InvalidObjectField(
 					null, "An API sort must be related to an API endpoint",
 					"an-api-sort-must-be-related-to-an-api-endpoint");
 			}
+
+			if (Validator.isNotNull(
+					_objectEntryHelper.getObjectEntry(
+						objectEntry.getCompanyId(),
+						StringBundler.concat(
+							"id ne '", objectEntry.getObjectEntryId(),
+							"' and r_apiEndpointToAPISorts_c_apiEndpointId eq ",
+							"'", apiEndpointId, "'"),
+						getObjectDefinitionExternalReferenceCode()))) {
+
+				ObjectDefinition objectDefinition =
+					_objectDefinitionLocalService.
+						getObjectDefinitionByExternalReferenceCode(
+							getObjectDefinitionExternalReferenceCode(),
+							objectEntry.getCompanyId());
+
+				User user = _userLocalService.getUser(objectEntry.getUserId());
+
+				String label = objectDefinition.getLabel(user.getLocale());
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					Collections.singletonList(label),
+					String.format(
+						"The API endpoint already has an associated %s", label),
```

Comment:
To maintain the consistency of the the message, I think that it should be:
`The API Endpoint already has an associated...` because the name of the label of the Object Definition is going to be in upper case too.

Reply:
Update it here and in the Language.properties

Reply:
Done! Thanks!